internal/status/immediatestatus.go

```diff
@@ -71,7 +74,7 @@ func (o *StatusObserver) getImmediateTopLevelStatusToReport() ImmediateTopLevelS
 		// Only report the latest active status for each goal state
 		goalStateKey := key.(types.GoalStateKey)
 		if goalStateKey.RuntimeSettingsState != "disabled" {
-			if value.(types.StatusItem) != (types.StatusItem{}) {
+			if !IsEmptyStatusItem(value.(types.StatusItem)) {
```

Comment:
Not following why this is necessary. Please at least add a comment stating why we need the DeepEqual call.